### PR TITLE
[.NET] Port patch fixes to release branch

### LINKED ISSUE #2148: Bot Builder v4 IgnoreEmptyItemsConverter causes an issue when ReadJson
# Platform
Bot Builder v4.0 (dotnet core)

# Author or host
web api (Bot Framework)

# Version of SDK
1.1

# Issue
When deserialize the DialogState which contains the AdaptiveCard, it fails with error "Error reading JArray from JsonReader at 

var array = JArray.Load(reader);. 

Error: Current JsonReader item is not an array: StartObject. Path 'DialogState.DialogStack.$values[0].State.dialogs.DialogStack.$values[0].State.options.Prompt.attachments.$values[0].content.action"

This caused by [JsonConverter(typeof(IgnoreEmptyItemsConverter<AdaptiveAction>))] attribute on AdaptiveCard which will be called while deserialization but it's not actually an array yet, as $value holds an array not action (nor body)

```json
"actions": {
    "$type": "System.Collections.Generic.List`1[[AdaptiveCards.ActionBase, AdaptiveCards]], System.Private.CoreLib",
    "$values": [
    {
        "$type": "AdaptiveCards.SubmitAction, AdaptiveCards",
        "type": "Action.Submit",
        "data": "Test",
        "title": "Test"
    }
    ]
}
```
We should try/catch JArray.Load

### LINKED ISSUE #5575: [.Net] Newtonsoft.Json 13.0.1 (latest) version explicitly not supported
# Platform

What platform is your issue or question related to? (Delete other platforms).

- [X] .NET HTML

# Author or host

Developer 

# Version of SDK

What version are you using? NuGet 1.2.1 -- (latest)

# Details

Latest nuget doesn't support (explicitly prevents) Newtonsoft.Json 13.0.1 (latest) version
image

### LINKED ISSUE #5963: [.NET][Templating] Null Substitution not working
# Target Application

Rendering in nodejs and/or Outlook, created using dotnet (AdaptiveCards.Templating 1.2.1, AdaptiveCards 2.7.1)

# Problem Description

Trying to render cards with null substitution, but not able to do so:

`string cardJson = template.Expand(data, (name) => "something");`

# Expected Outcome

The card should have the substitution text rendered (in this case "something")

# Actual Outcome

The tag is displayed like `${NullTagThatShouldBeReplaced}`

# Repro Steps

```
            AdaptiveCardTemplate template = new(text);
            var data = new
            {
                NullTagThatShouldBeReplaced= null
            };

            string cardJson = template.Expand(data, (name)=> "something");

            AdaptiveCardParseResult parseResult = AdaptiveCard.FromJson(cardJson);
            AdaptiveCard adaptiveCard = parseResult.Card;

            return adaptiveCard.ToJson();
```

# Schema Version

1.0

### LINKED ISSUE #6212: [Rendering] AdaptiveCards.Templating cannot be installed due to Newtonsoft.Json version clash
# Target Platform

Please list the platform(s) that your application is targeting and remove platforms you are not targeting:
- WPF

# Target Application
Teams

# Problem Description

It's impossible to add AdaptiveCards.Templating via NuGet install due to the following error:
NU1605	Detected package downgrade: Newtonsoft.Json from 12.0.3 to 11.0.2. Reference the package directly from the project to select a different version. 
 cron-test -> AdaptiveCards.Templating 1.2.1 -> AdaptiveExpressions 4.11.1 -> Newtonsoft.Json (>= 12.0.3) 
 cron-test -> AdaptiveCards.Templating 1.2.1 -> Newtonsoft.Json (>= 11.0.2 && <= 12.0.3)

It seems this is down to a newtonsoft.json version clash between AdaptiveCards.Templating (11.0.2) and one of it's dependencies AdaptiveExpressions (12.0.3).

# Screenshot

image

# Version of SDK
AdaptiveCards.Templating 1.2.1

### PR DESCRIPTION
**Pull request by paulcam206**
# Related Issue

Fixes #6212, Fixes #2148, Fixes #5963, Fixes #5575

# Description

Port of fixes in preparation of shipping `2.7.2` .NET package updates

# How Verified

* local build and tests, spot checking of rendering in WPF visualizer

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6226)

### CODE REVIEW

File: source/dotnet/Library/AdaptiveCards/docs/AdaptiveCards.md [hunk 1]
```diff
@@ -254,6 +275,7 @@
   - [FontSizes](#P-AdaptiveCards-Rendering-AdaptiveHostConfig-FontSizes 'AdaptiveCards.Rendering.AdaptiveHostConfig.FontSizes')
   - [FontTypes](#P-AdaptiveCards-Rendering-AdaptiveHostConfig-FontTypes 'AdaptiveCards.Rendering.AdaptiveHostConfig.FontTypes')
   - [FontWeights](#P-AdaptiveCards-Rendering-AdaptiveHostConfig-FontWeights 'AdaptiveCards.Rendering.AdaptiveHostConfig.FontWeights')
+  - [Headings](#P-AdaptiveCards-Rendering-AdaptiveHostConfig-Headings 'AdaptiveCards.Rendering.AdaptiveHostConfig.Headings')
```
Comment (RebeccaAnne) [inline source/dotnet/Library/AdaptiveCards/docs/AdaptiveCards.md:278]: Looks like some 1.5 stuff snuck in here?

File: source/dotnet/Library/AdaptiveCards/IgnoreEmptyItemsConverter.cs [hunk 2]
```diff
@@ -24,11 +24,23 @@ public override bool CanConvert(Type objectType)
...
         {
-            var array = JArray.Load(reader);
+            JToken jToken = JToken.Load(reader);
 
+            if (jToken is JObject jObject && jObject.HasValues)
+            {
+                jToken = jObject.GetValue("$values");
+            }
+
+            JArray jArray = new JArray();
```
Comment (jonmill) [inline source/dotnet/Library/AdaptiveCards/IgnoreEmptyItemsConverter.cs:34]: NIT: unnecessary allocation in the case that jToken is an array

**Review by jonmill — APPROVED** [review]
One minor nitpick but LGTM

File: source/dotnet/Library/AdaptiveCards/docs/AdaptiveCards.md (hunk 1 above)
Comment (paulcam206) [reply source/dotnet/Library/AdaptiveCards/docs/AdaptiveCards.md:278]: huh

**Comment by paulcam206** [comment]
@jwoo-msft - can you take a look at the iOS failures here? I obviously haven't changed anything relevant to iOS, but it would be good to know what's up here.

My *guess* is that it has to do with this change being against the older `release/schema-1.4` branch...

**Review by jwoo-msft — APPROVED** [review]

**Comment by jwoo-msft** [comment]
> @jwoo-msft - can you take a look at the iOS failures here? I obviously haven't changed anything relevant to iOS, but it would be good to know what's up here.
> 
> My _guess_ is that it has to do with this change being against the older `release/schema-1.4` branch...

Release/schema-1.4 base branch had this issue. It's due to inconsistency in the project file. Probably, a merge issue from main to 1.4 at some point. Fixed it and checked into your branch. FYI, main doesn't have this issue, should be fine going forward.

**Comment by paulcam206** [comment]
> > @jwoo-msft - can you take a look at the iOS failures here? I obviously haven't changed anything relevant to iOS, but it would be good to know what's up here.
> > My _guess_ is that it has to do with this change being against the older `release/schema-1.4` branch...
> 
> Release/schema-1.4 base branch had this issue. It's due to inconsistency in the project file. Probably, a merge issue from main to 1.4 at some point. Fixed it and checked into your branch. FYI, main doesn't have this issue, should be fine going forward.

awesome! thanks a bunch!

**Comment by paulcam206** [comment]
@jonmill - should we admin-power this merge in, or should I port the UWP and Android YML pipelines over to `release/schema-1.4`?

**Comment by jonmill** [comment]
@paulcam206 - if we don't see ourselves making patches to v1.4 then we can admin-push in. If we think we'll need to apply fixes later, then we should fix up the pipelines

**Comment by paulcam206** [comment]
@jonmill - got the pipeline definitions pulled in, but looks like UWP CI is failing on test cards (presumably due to deltas in our test infra or samples between `release/schema-1.4` and `main`). I think we should take this as "good enough for a .NET patch update" and take on backporting or fixing tests in the future if needed for a schema 1.4 UWP patch since these .NET changes have no bearing on UWP. What do you think?

**Comment by jonmill** [comment]
Sounds good to me!

**Comment by paulcam206** [comment]
@jonmill - I don't think I have permissions to force the checkin... care to do the honors? I believe in this case it would be appropriate to do a rebase-and-merge, but it's up to you :)

**Comment by jonmill** [comment]
Done!

**Pull request merged**